lib/fog/vsphere/requests/compute/vm_clone.rb

```diff
@@ -803,6 +804,7 @@ def modify_template_volumes_specs(vm_mob_ref, volumes)
                          device: template_volume }
             end
           end
+          new_volumes.map { |volume| volume.unit_number = volumes.index(volume) }
```

Comment:
I believe this is a bit too naive. I believe we have to skip unit number 7 since the pvscsi driver reserves SCSI ID 7.